File: app/src/main/java/com/gmail/cristiandeives/motofretado/http/jsonapi/BusAttributes.java
package com.gmail.cristiandeives.motofretado.http.jsonapi;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;
import com.gmail.cristiandeives.motofretado.http.Bus;

import java.util.Date;

class BusAttributes {
    double latitude;
    double longitude;
    @SerializedName("created_at")
    @Nullable Date createdAt;
    @SerializedName("updated_at")
    @Nullable Date updatedAt;

    static @NonNull BusAttributes parse(@NonNull Bus bus) {
        BusAttributes attributes = new BusAttributes();

        attributes.createdAt = bus.createdAt;
        attributes.updatedAt = bus.updatedAt;
        attributes.latitude = bus.latitude;
        attributes.longitude = bus.longitude;

        return attributes;
    }
}

File: app/src/main/java/com/gmail/cristiandeives/motofretado/http/jsonapi/BusAttributes.kt
package com.gmail.cristiandeives.motofretado.http.jsonapi

import com.gmail.cristiandeives.motofretado.http.Bus
import com.google.gson.annotations.SerializedName
import java.util.Date

internal data class BusAttributes(
        val latitude: Double?,
        val longitude: Double?,
        @SerializedName("created_at") val createdAt: Date?,
        @SerializedName("updated_at") val updatedAt: Date?
) {
    constructor(bus: Bus) : this(
            bus.latitude,
            bus.longitude,
            bus.createdAt,
            bus.updatedAt
    )
}